### PR TITLE
Add special case for normalizing gmail addresses

### PR DESCRIPTION
**Pull request by gustaveen**
## Describe your changes
Add a conditional field to user data, `gem`, that passes a gmail address formatted according to Googles documentation https://developers.google.com/google-ads/api/docs/conversions/enhanced-conversions/web#python

> Remove all periods (.) that precede the domain name in gmail.com and googlemail.com email addresses

## Justify why they are needed
Fix user data tracking

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by gustaveen** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4102?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4102** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4102?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4101** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4101?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gustaveen and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/services/Tracking/Tracking.ts [hunk 1]
```diff
@@ -389,5 +389,19 @@ export const normalizeEmail = (value: unknown) => {
...
+  if (!normalizedEmail) return undefined
+
+  // Remove all periods (.) that precede the domain name
+  // in gmail.com and googlemail.com email addresses
+  // https://developers.google.com/google-ads/api/docs/conversions/enhanced-conversions/web#python
+  const gmailRegex = /^(gmail|googlemail)\.com\s*/
+  const emailParts = normalizedEmail.split('@')
+  const isGmail = emailParts.length > 1 && gmailRegex.test(emailParts[1])
+  if (isGmail) {
+    emailParts[0] = emailParts[0].replace('.', '')
```
Comment (alebedev) [inline apps/store/src/services/Tracking/Tracking.ts:402]: `.replaceAll` ?

**Review by alebedev — APPROVED** [review]

**Review by guilhermespopolin — APPROVED** [review]
🚀

**Comment by gustaveen** [comment]
### Merge activity

* **Apr 5, 7:54 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4102) rebased this pull request as part of a merge.

**Pull request merged**